在maxpool函数中加入dilation参数

Qin-sx opened the pull request:
<!-- TemplateReference: https://github.com/PaddlePaddle/Paddle/wiki/PULL-REQUEST-TEMPLATE--REFERENCE -->
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->

### PR Category
<!-- One of [ User Experience | Execute Infrastructure | Operator Mechanism | CINN | Custom Device | Performance Optimization | Distributed Strategy | Parameter Server | Communication Library | Auto Parallel | Inference | Environment Adaptation ] -->

User Experience

### PR Types
<!-- One of [ New features | Bug fixes | Improvements | Performance | BC Breaking | Deprecations | Docs | Devs | Not User Facing | Security | Others ] -->

Improvements

### Description
<!-- Describe what you’ve done -->

在maxpool函数中加入了dilation参数

Qin-sx committed:
added dilation para in maxpool funcs

Qin-sx committed:
updated for gpu

	modified:   paddle/phi/kernels/funcs/pooling.cu
	modified:   paddle/phi/kernels/funcs/pooling.h

zhwesky2010 reviewed (commented):
目前CI还没通过，另外onednn和xpu的OP是不是也需要一起测试下，这几个单测：

```
test/mkldnn/test_pool2d_mkldnn_op.py
test/xpu/test_pool2d_op_xpu.py
test/xpu/test_pool3d_op_xpu.py
```

Qin-sx commented:
> 目前CI还没通过，另外onednn和xpu的OP是不是也需要一起测试下，这几个单测：
> 
> ```
> test/mkldnn/test_pool2d_mkldnn_op.py
> test/xpu/test_pool2d_op_xpu.py
> test/xpu/test_pool3d_op_xpu.py
> ```

嗯，收到，我目前还在看CINN这部分怎么修改，这块还不太熟悉

Qin-sx committed:
updated for CINN

	modified:   paddle/cinn/hlir/dialect/operator/ir/ops.yaml
	modified:   paddle/cinn/hlir/dialect/operator/transforms/cinn_to_pd_util.cc
	modified:   paddle/cinn/hlir/dialect/operator/transforms/pd_to_cinn_pass.cc
	modified:   paddle/cinn/hlir/op/nn.cc
	modified:   paddle/cinn/hlir/pe/nn.cc
	modified:   paddle/cinn/hlir/pe/nn.h

Qin-sx committed:
Merge branch 'develop' into opt-maxpool

Qin-sx committed:
updated PoolImpl for CINN

	modified:   paddle/cinn/hlir/pe/nn.cc

Qin-sx committed:
update pool1d and pool3d for CINN

	modified:   paddle/cinn/hlir/op/nn.cc
	modified:   paddle/cinn/hlir/pe/nn.h

Qin-sx committed:
updated for cinn

	modified:   paddle/cinn/hlir/pe/nn.h
	modified:   test/cinn/test_op_nn.py

Qin-sx committed:
updated for pool3d in tensorRT

	modified:   paddle/fluid/inference/tensorrt/plugin/pool3d_op_plugin.cu
	modified:   paddle/fluid/inference/tensorrt/plugin/pool3d_op_plugin.h

Qin-sx committed:
update for tensorRT

	modified:   paddle/fluid/inference/tensorrt/plugin/pool3d_op_plugin.cu
	modified:   paddle/fluid/inference/tensorrt/plugin/pool3d_op_plugin.h

Qin-sx committed:
update for trt

	modified:   paddle/fluid/inference/tensorrt/plugin/pool_op_plugin.cu
	modified:   paddle/fluid/inference/tensorrt/plugin/pool_op_plugin.h

Qin-sx committed:
update for trt

	modified:   paddle/fluid/inference/tensorrt/convert/pool2d_op.cc
	modified:   paddle/fluid/inference/tensorrt/convert/pool3d_op.cc
	modified:   paddle/fluid/inference/tensorrt/plugin/pool_op_plugin.cu
	modified:   paddle/fluid/inference/tensorrt/plugin/pool_op_plugin.h
	modified:   paddle/phi/kernels/funcs/pooling.cu
	modified:   paddle/phi/kernels/funcs/pooling.h

Qin-sx committed:
codestyle

	modified:   paddle/cinn/hlir/op/nn.cc

Qin-sx committed:
update for hip

	modified:   paddle/phi/kernels/gpudnn/pool_grad_kernel.cu
	modified:   paddle/phi/kernels/pool_grad_kernel.h

Qin-sx committed:
update for hip

	modified:   paddle/phi/kernels/gpudnn/pool_grad_kernel.cu
	modified:   paddle/phi/kernels/pool_grad_kernel.h

Qin-sx committed:
updated for cudnn

	modified:   paddle/phi/kernels/fusion/gpu/max_pool2d_v2_grad_kernel.cu
	modified:   paddle/phi/kernels/fusion/gpu/max_pool2d_v2_kernel.cu

Qin-sx committed:
updated for test

	modified:   test/legacy_test/test_imperative_layers.py

Qin-sx committed:
updated for test

	modified:   test/legacy_test/test_imperative_layers.py

Qin-sx committed:
only added para dilation in xpu functions

	modified:   paddle/phi/kernels/xpu/pool_grad_kernel.cc
	modified:   paddle/phi/kernels/xpu/pool_kernel.cc

Qin-sx committed:
Merge branch 'develop' into opt-maxpool

Qin-sx committed:
compiled

	modified:   paddle/fluid/pir/drr/src/ir_operation_factory.cc
	modified:   paddle/phi/infermeta/backward.h
	modified:   paddle/phi/infermeta/unary.h
	modified:   paddle/phi/kernels/funcs/pooling.cc
	modified:   paddle/phi/kernels/funcs/pooling.cu
	modified:   paddle/phi/kernels/gpudnn/pool_kernel.cu
	modified:   paddle/phi/kernels/impl/pool_grad_kernel_impl.h
	modified:   paddle/phi/kernels/impl/pool_kernel_impl.h

Qin-sx committed:
format

	modified:   paddle/phi/infermeta/unary.cc
	modified:   paddle/phi/kernels/funcs/pooling.cc
	modified:   paddle/phi/kernels/funcs/pooling.cu
	modified:   paddle/phi/kernels/impl/pool_kernel_impl.h

Qin-sx committed:
format

	modified:   test/legacy_test/test_pool_max_op.py

Qin-sx committed:
format

	modified:   paddle/phi/ops/yaml/legacy/static_backward.yaml
	modified:   paddle/phi/ops/yaml/legacy/static_ops.yaml

Qin-sx committed:
format

	modified:   paddle/fluid/pir/transforms/tensorrt/trt_op_marker_pass.cc
